src/EnergyPlus/RuntimeLanguageProcessor.cc

```diff
@@ -2151,24 +2171,43 @@ namespace RuntimeLanguageProcessor {
 			for ( OperandNum = 1; OperandNum <= ErlExpression( ExpressionNum ).NumOperands; ++OperandNum ) {
 				Operand( OperandNum ) = ErlExpression( ExpressionNum ).Operand( OperandNum );
 				if ( Operand( OperandNum ).Type == ValueExpression ) {
-					Operand( OperandNum ) = EvaluateExpression( Operand( OperandNum ).Expression ); //recursive call
+					Operand( OperandNum ) = EvaluateExpression( Operand( OperandNum ).Expression, seriousErrorFound ); //recursive call
 				} else if ( Operand( OperandNum ).Type == ValueVariable ) {
-					Operand( OperandNum ) = ErlVariable( Operand( OperandNum ).Variable ).Value;
+					if ( ErlVariable( Operand( OperandNum ).Variable ).Value.initialized ) { // check that value has been initialized
+						Operand( OperandNum ) = ErlVariable( Operand( OperandNum ).Variable ).Value;
+					} else { // value has never been set
+						ReturnValue.Type = ValueError;
+						ReturnValue.Error = "EvaluateExpression: Variable = '" + ErlVariable( Operand( OperandNum ).Variable ).Name + "' used in expression has not been initialized!" ;
+						if ( ! DoingSizing && ! KickOffSimulation && ! EMSManager::FinishProcessingUserInput ) {
+
+							//check if this is an arg in CurveValue,
+							if ( ErlExpression( ExpressionNum ).Operator != FuncCurveValue ) { // padding the argument list for CurveValue is too common to fatal on.  only reported to EDD
```

Comment:
So the point of this trap is because `@CurveValue` could be given a variable number of arguments, and you are protecting users that may leave stray arguments?  

I'm not 100% against this, but my inclination would definitely be to warn/error here.  I'm curious why you think it is so common, or if there was some conversation you can point me to justifying it.